### DISCUSSION TITLE
BindingContext changes after being added to Canvas

### DISCUSSION THREAD

**Discussion by sferhah** [taublast/DrawnUi#119] [Q&A] [answered]
Tell me if I am wrong:

**Expected Behavior:** The BindingContext of the Skia control should not be reset when added to the visual tree.
**Actual Behavior:** The BindingContext of the Skia control changes to parent's BindingContext .

**Code Sample:**
```
public record MyItem(string Name, string Description, string[] SubItems, string? SelectedItem);

public class MyPage : ContentPage
{
    public MyPage()
    {
        BindingContext = this;

        var item = new MyItem("Name", "Description", new[] { "a", "b", "c" }, "c");

        var button = new SkiaButton
        {
            Text = "Test",
            BindingContext = item
        };

        // BindingContext is MyItem
        var context1 = button.BindingContext;

        Content = new Canvas
        {
            Content = button
        };

        // BindingContext changes to "MyPage"
        var context2 = button.BindingContext;
        // context1 should be equal to context2
    }
}
```

**Comment by taublast** [comment, marked as answer]
Hi @sferhah. Maybe i am wrong but this is intended behaviour you can adjust for your needs.

1. Parents always propagate their BindingContext to children, when child has its Parent property set. This is by design.
2. You can change this for a specific control if you need a different behavior:

I see your main problem occurs when you add your layout to canvas, and canvas sets its context. so you can have a custom wrapper denying this context:

Would suggest to always use a wrapper layout inside canvas even you want to use just one control, like button or label. Why? For potential overlays and to be able to cache your control when it has some surrounding stuff like shadows. In this case you would add a margin to your control with shadows and cache the wrapper, so the whole will not be redrawn everytime MAUI wants it too, it would just redraw cache in no time.

```csharp
 public class Wrapper : SkiaLayout
 {
     public override void SetInheritedBindingContext(object context)
     {
         //i deny your context:

         //wll not be calling base
         //base.SetInheritedBindingContext(context);
     }
 }
```

Then wrap your button inside the Wrapper, do something, then add wrapper to canvas.

**Comment by taublast** [comment]
Please feel free to ask more questions and raise up issues, would be glad to help. Will be closing this one now.

**Comment by sferhah** [comment]
@taublast, 

I tried migrating some code to DrawnUi and encountered the issue. I expected it to behave the same as with classic Maui controls. I fixed it by assigning the BindingContext after adding the view. It seemed odd, so I posted it here, thinking it might not be intended.

**Reply by taublast** [reply]
Sure, MAUI might do a check like if BindingContext!=null then adopt parent BindingContext. It's their debatable choice.
